bundles/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/handler/SpeedTestHandler.java

```diff
@@ -132,7 +130,17 @@ public void onError(final @Nullable SpeedTestError testError, final @Nullable St
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMessage);
             freeRefreshTask();
             return;
-        } else if (SpeedTestError.SOCKET_TIMEOUT.equals(testError) || SpeedTestError.SOCKET_ERROR.equals(testError)
+        } else if (SpeedTestError.SOCKET_TIMEOUT.equals(testError)) {
+            timeouts--;
+            if (timeouts <= 0) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Max timeout count reached");
+                freeRefreshTask();
```

Comment:
I was checking your changes when I realized this fix doesn't actually fix the original report it only delays the problem. The problem is the binding will stop refresh at some point. The only fix this was to restart the bundle. With the added changes to handleCommand that also can fix it. However. I was wondering why it should be stopped at all? why not just continue trying forever? It can just set the state to offline, but still continue the refresh tasks and set the ThingStatus.ONLINE in `onComplete` if it was `OFFLINE`?

Reply:
It depends. It can fix in case of ponctual timeouts. It will not if timeouts are frequent (and in this case I think the best way is to find another endpoint for the speedtest). But you're right, there is no real reason to put it offline.